core/perf_test/PerfTest_MallocFree.cpp

```diff
@@ -0,0 +1,94 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+#include <benchmark/benchmark.h>
+#include "Benchmark_Context.hpp"
+
+namespace Test {
+
+static constexpr size_t STRIDE = 1024;  // touch the allocation with this stride
+
+// when the time will be recorded
+enum class When { after_malloc, after_touch, after_free };
+
+static void Impl(benchmark::State& state, const bool touch, const When& when) {
+  const size_t N2 = std::pow(2, state.range(0));
```

Comment:
Is there a good reason for doing this inside the body of the benchmark (formatting of the output?) rather than via the range argument?
In other words why are we using `DenseRange` and then taking powers of two instead of going with a regular Range+RangeMultiplier

Doc here https://github.com/google/benchmark/blob/main/docs/user_guide.md#passing-arguments